cmd/parca/main.go

```diff
@@ -22,36 +22,42 @@ import (
 	"github.com/common-nighthawk/go-figure"
 	"github.com/go-kit/log/level"
 	"github.com/parca-dev/parca/pkg/parca"
+	"github.com/parca-dev/parca/pkg/server"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 var (
-	version string
-	commit  string
-	date    string
-	builtBy string
+	version string = "dev"
+	commit  string = "dev"
 )
 
 func main() {
 	ctx := context.Background()
 	flags := &parca.Flags{}
+
 	kong.Parse(flags)
 
+	if flags.Version {
+		fmt.Printf("parca, version %s (commit: %s)\n", version, commit)
+		return
+	}
+
 	serverStr := figure.NewColorFigure("Parca", "roman", "cyan", true)
 	serverStr.Print()
 
 	logger := parca.NewLogger(flags.LogLevel, parca.LogFormatLogfmt, "parca")
 	level.Debug(logger).Log("msg", "parca initialized",
 		"version", version,
 		"commit", commit,
-		"date", date,
-		"builtBy", builtBy,
 		"config", fmt.Sprint(flags),
 	)
 
 	registry := prometheus.NewRegistry()
 
-	err := parca.Run(ctx, logger, registry, flags)
+	err := parca.Run(ctx, logger, registry, flags, &server.UIData{
```

Comment:
My apologies, I wasn't clear in my previous comment, I also meant that at this point it shouldn't even be "UIData" but just version information, and where rendering happens it can be transformed into a different datastructure (that currently happens to be identical to a "VersionInfo" struct for example).

Reply:
Ah thanks - thats a good idea, changed